ejb3/src/test/java/org/jboss/as/ejb3/delivery/EJBBoundMdbDeliveryMetaDataParserTestCase.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.jboss.as.ejb3.delivery;
+
+import org.jboss.as.ejb3.delivery.parser.EJBBoundMdbDeliveryMetaDataParser;
+import org.jboss.as.ejb3.delivery.parser.EjbBoundMdbDeliveryMetaDataSchema;
+import org.jboss.metadata.ejb.parser.jboss.ejb3.JBossEjb3MetaDataParser;
+import org.jboss.metadata.ejb.parser.spec.AbstractMetaDataParser;
+import org.jboss.metadata.parser.util.MetaDataElementParser;
+import org.jboss.metadata.property.PropertyReplacer;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamReader;
+import java.io.InputStream;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+
+public class EJBBoundMdbDeliveryMetaDataParserTestCase {
+
+    private JBossEjb3MetaDataParser parser;
+    private PropertyReplacer mockReplacer;
+
+    @Before
+    public void init() {
+        Map<String, AbstractMetaDataParser<?>> parsers = new HashMap<>();
+        for (EjbBoundMdbDeliveryMetaDataSchema schema : EnumSet.allOf(EjbBoundMdbDeliveryMetaDataSchema.class)) {
+            parsers.put(schema.getNamespace().getUri(), new EJBBoundMdbDeliveryMetaDataParser(schema));
+        }
+        parser = new JBossEjb3MetaDataParser(parsers);
+
+        mockReplacer = new PropertyReplacer() {
+            @Override
+            public String replaceProperties(String s) {
+                return s;
+            }
+        };
+    }
+
+    @Test
+    public void test_3_0() throws Exception {
+        try (InputStream is = this.getClass().getResourceAsStream("jboss-ejb3-delivery-3.0.xml")) {
+            XMLStreamReader reader = XMLInputFactory.newInstance().createXMLStreamReader(is);
+            parser.parse(reader, new MetaDataElementParser.DTDInfo(), mockReplacer);
+        }
+    }
+
+    @Test
+    public void test_2_0() throws Exception {
+        try (InputStream is = this.getClass().getResourceAsStream("jboss-ejb3-delivery-2.0.xml")) {
+            XMLStreamReader reader = XMLInputFactory.newInstance().createXMLStreamReader(is);
+            parser.parse(reader, new MetaDataElementParser.DTDInfo(), mockReplacer);
+        }
+    }
+
+    @Test
+    public void test_1_1() throws Exception {
+        try (InputStream is = this.getClass().getResourceAsStream("jboss-ejb3-delivery-1.1.xml")) {
+            XMLStreamReader reader = XMLInputFactory.newInstance().createXMLStreamReader(is);
+            parser.parse(reader, new MetaDataElementParser.DTDInfo(), mockReplacer);
+        }
+    }
+
+    @Test
+    public void test_1_0() throws Exception {
+        try (InputStream is = this.getClass().getResourceAsStream("jboss-ejb3-delivery-1.0.xml")) {
+            XMLStreamReader reader = XMLInputFactory.newInstance().createXMLStreamReader(is);
+            parser.parse(reader, new MetaDataElementParser.DTDInfo(), mockReplacer);
+        }
+    }
```

Comment:
Please parameterize this test instead.
e.g.
```java
@RunWith(Parameterized.class)
public class EJBBoundMdbDeliveryMetaDataParserTestCase {
    @Parameters
    public static Iterable<EjbBoundMdbDeliveryMetaDataSchema> parameters() {
        return EnumSet.allOf(EjbBoundMdbDeliveryMetaDataSchema.class);
    }

    private final EjbBoundMdbDeliveryMetaDataSchema schema;

    public EJBBoundMdbDeliveryMetaDataParserTestCase(EjbBoundMdbDeliveryMetaDataSchema schema) {
        this.schema = schema;
    }
    //...
}
```

Reply:
@pferraro I have parametrisized the test.